library/classes/Address.class.php

```diff
@@ -52,11 +52,12 @@ static function factory_address($foreign_id = "")
         }
 
         $a = new Address();
-        $sql = "SELECT id FROM  " . $a->_table . " WHERE foreign_id " .$foreign_id ;
+        $sql = "SELECT id FROM  " . $a->_table . " WHERE foreign_id = ?";
```

Comment:
Need to also escape the table here, which is done using a functions that uses a whitelisting strategy(ie. dies if table does not exist):
```
$sql = "SELECT id FROM  " . escape_table_name($a->_table) . " WHERE foreign_id = ?";
```